File: app/src/main/java/com/waveneuro/ui/user/login/LoginCommand.java
package com.waveneuro.ui.user.login;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;

import com.asif.abase.base.NavigationCommand;
import com.asif.abase.injection.qualifier.ActivityContext;

import javax.inject.Inject;

public class LoginCommand extends NavigationCommand {

    private final Context mContext;

    @Inject
    public LoginCommand(@ActivityContext Context mContext) {
        this.mContext = mContext;
    }

    @Override
    public void navigate() {
        Intent intent = new Intent(this.mContext, LoginActivity.class);
        this.mContext.startActivity(intent);
        ((Activity) this.mContext).finish();
    }
}

File: app/src/main/java/com/waveneuro/ui/user/login/LoginCommand.kt
package com.waveneuro.ui.user.login

import android.app.Activity
import android.content.Context
import android.content.Intent
import com.asif.abase.base.NavigationCommand
import com.asif.abase.injection.qualifier.ActivityContext
import javax.inject.Inject

class LoginCommand @Inject constructor(@param:ActivityContext private val mContext: Context) :
    NavigationCommand() {
    override fun navigate() {
        val intent = Intent(mContext, LoginActivity::class.java)
        intent.flags =
            Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
        mContext.startActivity(intent)
        (mContext as Activity).finish()
    }
}